src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -470,6 +478,7 @@ internal void HandleWorkerError(Exception exc)
             {
                 if (!_disposed)
                 {
+                    _startLatencyMetric?.Dispose();
```

Comment:
This should also be done in the Dispose path (same for the `_startSubscription` instance).

This does put you in a position where you could potentially call dispose on this type twice, but we shouldn't rely just on the process exit path to do this, principally since there could be an issue killing the process or a race (since we don't wait for the process to exit there) where you'd end up setting the `_disposed` flag. Looking at the current implementation of  `DisposableEvent`, this also means you'd need to update that type so multiple dispose calls are not an issue.